controllers/mapController.js

```diff
@@ -110,7 +115,90 @@ const addInvitedUser = async (req, res, next) => {
   }
 };
 
+const inviteNewMember = async (req, res, next) => {
+  const { id } = req.params;
+  const { email } = req.body;
+
+  try {
+    const currentMap = await Map.findById(id).exec();
+    const newUser = await User.findOne({ email }).exec();
+
+    const userId = newUser._id;
+    const verifyInvitedEmail = currentMap.invitationList
+      .map(user => user.email)
+      .some(invitedEmail => invitedEmail === email);
+
+    if (!newUser) {
+      res.json({
+        result: "등록되지 않은 유저입니다.",
+      });
+
+      return;
+    }
+
+    if (currentMap.members.some(member => member.equals(userId))) {
+      res.json({
+        result: "이미 같은 그룹 멤버입니다.",
+      });
+
+      return;
+    }
+
+    if (verifyInvitedEmail) {
+      res.json({
+        result: "이미 초대 메일을 보낸 유저입니다.",
+      });
+
+      return;
+    }
+
+    const invitationToken = jwt.sign(
+      { email },
+      process.env.INVITATION_SECRET_KEY,
+      {
+        expiresIn: "2d",
+      }
+    );
+
+    const invitationUrl = `http://localhost:3000/my-travels/${id}/invitation/${invitationToken}`;
+
+    const transporter = nodemailer.createTransport({
+      service: "gmail",
+      auth: {
+        user: INVITATION_MAIL,
+        pass: INVITATION_PASSWORD,
+      },
+    });
+
+    const sendMail = transporter.sendMail({
+      from: INVITATION_MAIL,
+      to: email,
+      subject: "[OCN] 그룹 초대 메일",
+      html: `<p> 아래 링크를 누르면 그룹으로 초대 됩니다 </p> <a href=${invitationUrl} >초대링크</a>`,
+    });
+
+    currentMap.invitationList.push({
+      email,
+      token: invitationToken,
+    });
+
+    await Promise.all([sendMail, currentMap.save()]);
+
```

Comment:
서영님~ 현재 185번 줄에서 `sendMail` 을 실행한다는 의미로 작성해주신 것 같은데 위의 173~183번 줄에서 이미 실행이 되어버리는 것 같이 보이는 것 같습니다! 

혹시 제가 생각하는 것이 맞다면 sendMail 에 들어가는 object 옵션을 따로 변수로 관리해주고 아래에서 함께 실행해주는 방식으로 수정해주면 서영님께서 의도하신 것처럼 실행시킬 수 있을 것 같아요~!

```suggestion
    const option = {
      from: INVITATION_MAIL,
      to: email,
      subject: "[OCN] 그룹 초대 메일",
      html: `<p> 아래 링크를 누르면 그룹으로 초대 됩니다 </p> <a href=${invitationUrl} >초대링크</a>`,
    });

    currentMap.invitationList.push({
      email,
      token: invitationToken,
    });

    await Promise.all([transporter.sendMail(option), currentMap.save()]);

```

Reply:
훨씬 깔끔하네용ㅎㅎ! 👍